package/network/utils/linux-atm/patches/800-format.patch

```diff
@@ -0,0 +1,166 @@
+ 	exit(1);
+     }
+     wrote = write(fd,data,size);
+--- a/src/mpoad/lecs.c
++++ b/src/mpoad/lecs.c
+@@ -157,7 +157,7 @@ static int get_reply(int fd, char *buff,
+                 /* Sampo-Add: end */
+         }
+         if (end_of_tlvs - tlvs != 0)
+-                printf("mpcd: lecs.c: get_reply: ignoring %d bytes of trailing TLV carbage\n",
++                printf("mpcd: lecs.c: get_reply: ignoring %" PRIdPTR " bytes of trailing TLV carbage\n",
+                        end_of_tlvs - tlvs);
+         return 1;
+ }
+--- a/src/switch/swc.c
++++ b/src/switch/swc.c
+@@ -28,7 +28,7 @@ static void dialog(int s,SWC_MSG *msg)
+ 	exit(1);
+     }
+     if (size != sizeof(*msg)) {
+-	fprintf(stderr,"bad write: %d != %d\n",size,sizeof(*msg));
++	fprintf(stderr,"bad write: %d != %zu\n",size,sizeof(*msg));
+ 	exit(1);
+     }
+     size = read(s,msg,sizeof(*msg));
+@@ -37,7 +37,7 @@ static void dialog(int s,SWC_MSG *msg)
+ 	exit(1);
+     }
+     if (size != sizeof(*msg)) {
+-	fprintf(stderr,"bad read: %d != %d\n",size,sizeof(*msg));
++	fprintf(stderr,"bad read: %d != %zu\n",size,sizeof(*msg));
+ 	exit(1);
+     }
+ }
+--- a/src/test/isp.c
++++ b/src/test/isp.c
+@@ -37,7 +37,7 @@ void send_msg(const struct atmsvc_msg *m
+     wrote = write(sock,msg,sizeof(*msg));
+     if (wrote == sizeof(*msg)) return;
+     if (wrote < 0) perror("write");
+-    else fprintf(stderr,"bad write: %d != %d\n",wrote,sizeof(*msg));
++    else fprintf(stderr,"bad write: %d != %zd\n",wrote,sizeof(*msg));
+     exit(1);
+ }
+ 
+@@ -49,7 +49,7 @@ void recv_msg(struct atmsvc_msg *msg)
+     got = read(sock,msg,sizeof(*msg));
+     if (got == sizeof(*msg)) return;
+     if (got < 0) perror("read");
+-    else fprintf(stderr,"bad read: %d != %d\n",got,sizeof(*msg));
++    else fprintf(stderr,"bad read: %d != %zd\n",got,sizeof(*msg));
+     exit(1);
+ }
+ 
+--- a/src/test/ttcp.c
++++ b/src/test/ttcp.c
+@@ -836,13 +836,13 @@ prusage(r0, r1, e, b, outp)
+ 
+ 		case 'U':
+ 			tvsub(&tdiff, &r1->ru_utime, &r0->ru_utime);
+-			sprintf(outp,"%ld.%01ld", tdiff.tv_sec, tdiff.tv_usec/100000);
++			sprintf(outp,"%" PRIu64 ".%01" PRIu64 , (uint64_t)tdiff.tv_sec, (uint64_t)tdiff.tv_usec/100000);
+ 			END(outp);
+ 			break;
+ 
+ 		case 'S':
+ 			tvsub(&tdiff, &r1->ru_stime, &r0->ru_stime);
+-			sprintf(outp,"%ld.%01ld", tdiff.tv_sec, tdiff.tv_usec/100000);
++			sprintf(outp,"%" PRIu64 ".%01" PRIu64, (uint64_t)tdiff.tv_sec, (uint64_t)tdiff.tv_usec/100000);
+ 			END(outp);
+ 			break;
+ 
+@@ -864,19 +864,19 @@ prusage(r0, r1, e, b, outp)
+ 			break;
+ 
+ 		case 'X':
+-			sprintf(outp,"%ld", t == 0 ? 0 : (r1->ru_ixrss-r0->ru_ixrss)/t);
++			sprintf(outp, PRIu64, t == 0 ? 0 : (uint64_t)(r1->ru_ixrss-r0->ru_ixrss)/t);
+ 			END(outp);
+ 			break;
+ 
+ 		case 'D':
+-			sprintf(outp,"%ld", t == 0 ? 0 :
+-			    (r1->ru_idrss+r1->ru_isrss-(r0->ru_idrss+r0->ru_isrss))/t);
++			sprintf(outp, PRIu64, t == 0 ? 0 :
++			    (uint64_t)(r1->ru_idrss+r1->ru_isrss-(r0->ru_idrss+r0->ru_isrss))/t);
```

Comment:
what type was used here before you added the cast? Adding the old error message to the comment should be fine.

Reply:
long. Now it’s long long.

Reply:
if it was long then, the change is not needed.

Reply:
Sorry, under 32-bit it's long long. Under 64-bit it's long.